frontend/src/contexts/apolloClientContext.tsx

```diff
@@ -8,33 +8,49 @@ import { setContext } from "@apollo/client/link/context";
 import { ReactNode, useContext, useEffect, useState } from "react";
 import { User, UserContext } from "../contexts/userContext";
 import { Roles } from "../utils/types";
+import { auth } from "../../firebaseConfig"; // Ensure this is the path to your Firebase configuration
 
 export const GRAPHQL_URI = "http://127.0.0.1:9191/v1/graphql";
 
 const httpLink = createHttpLink({
   uri: GRAPHQL_URI,
 });
 
-const createAuthLink = (user?: User) =>
-  setContext((_, { headers }) => {
+const createAuthLink = (user?: User, tokenBypass?: string) =>
```

Comment:
Can you change it to

```suggestion
const createAuthLink = (user: User | undefined, tokenBypass: string | undefined) =>
```

The difference it that with this change `createAuthLink()` call is not possible, you always have to pass two parameters.